patches/common/chromium/crashpad_pid_check.patch

```diff
@@ -0,0 +1,33 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Cheng Zhao <zcbenz@gmail.com>
+Date: Tue Jun 4 11:30:12 JST 2019
+Subject: crashpad_pid_check.patch
+
+When both browser process and renderer process are connecting to the pipe,
+the API may return the PID of browser process as real_pid, which is different
+from the PID of renderer process.
+
+I don't understand why Chromium does not have this issue, and we should
+look into it in future.
+
+diff --git a/third_party/crashpad/crashpad/util/win/exception_handler_server.cc b/third_party/crashpad/crashpad/util/win/exception_handler_server.cc
+index 2593ff2de032..e89b8ff675be 100644
+--- a/third_party/crashpad/crashpad/util/win/exception_handler_server.cc
++++ b/third_party/crashpad/crashpad/util/win/exception_handler_server.cc
+@@ -448,9 +448,16 @@ bool ExceptionHandlerServer::ServiceClientConnection(
+     DWORD real_pid = 0;
+     if (get_named_pipe_client_process_id(service_context.pipe(), &real_pid) &&
+         message.registration.client_process_id != real_pid) {
++      // Electron: When both browser process and renderer process are connecting
++      // to the pipe, the API may return the PID of browser process as real_pid,
++      // which is different from the PID of renderer process.
++      //
++      // I don't understand why Chromium does not have this issue.
```

Comment:
This is somewhat concerning, what are the implications of the incorrect PID being returned here?

Reply:
Chrome uses `GetNamedPipeClientProcessId` to check if the PID passed via pipe is actually the process connecting the pipe. However there may be multiple processes connecting the pipe and `GetNamedPipeClientProcessId` only returns the first PID, which ends up with the check here.

The PID returned by `GetNamedPipeClientProcessId` is only used for the check here, it does not affect other parts.

I guess Chrome does not have this problem because they put crashReporter code in `chrome_elf.dll`.

Reply:
Hm, I'm not sure I understand how putting the code in a different dll would affect the order of processes that connect to the pipe or the process IDs returned by `GetNamedPipeClientProcessId`. Could you expand on the `chrome_elf.dll` theory?

Reply:
It is a wild guess, I don't really understand why this check passes in Chromium.